src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -610,6 +612,9 @@ private static bool StringRequiresHtmlDecoding(string s)
         // Internal struct to facilitate URL decoding -- keeps char buffer and byte buffer, allows appending of either chars or bytes
         private struct UrlDecoder
         {
+            public bool _containsUnsafe;
+            public bool _containsSpaces;
```

Comment:
I'd prefer it if you kept them as a single local now.  If a subsequent change needs them separated, then that change can do it.  Otherwise, if for example that change never materializes or gets merged, we're left with debt.